### PR TITLE
build: add integration testing for marketing analytics SDK

### PR DESCRIPTION
**Pull request by kevinpagtakhan**
### Summary

Adds integration testing (plugins + browser SDK). The tests verifies the following cases:

* Track attribution only
* Track page view on attribution
* Track page view on load
* Track attribution and page view on load separately

### Checklist

* [x] Does your PR title have the correct [title format](https://github.com/amplitude/Amplitude-TypeScript/blob/main/CONTRIBUTING.md#pr-commit-title-conventions)?
* Does your PR have a breaking change?:  No

### CODE REVIEW

**Review by dnshi — APPROVED** [review]

File: packages/analytics-ma-browser/test/e2e.test.ts [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+        expect(requestBody).toEqual({
+          api_key: 'API_KEY',
+          events: [
+            {
+              device_id: expectString,
+              event_id: 0,
+              event_properties: {
+                page_location: 'http://localhost/',
+                page_path: '/',
+                page_title: '',
```
Comment (AlecBoutin) [inline packages/analytics-ma-browser/test/e2e.test.ts:130]: would expect `page_domain` and `page_url` to also appear here
image

**Review by AlecBoutin — APPROVED** [review]
Looks good!

File: packages/analytics-ma-browser/test/e2e.test.ts (hunk 1 above)
Comment (kevinpagtakhan) [reply packages/analytics-ma-browser/test/e2e.test.ts:130]: correct it should be there. let me check. might be because it has not been rebased yet

Comment (kevinpagtakhan) [reply packages/analytics-ma-browser/test/e2e.test.ts]: i updated the test and these field are being asserted including the additional click ids

**Pull request merged**